dynamodb/db.go

```diff
@@ -47,27 +48,32 @@ func (d ddbBucketStatePrimaryKey) KeySchema() []*dynamodb.KeySchemaElement {
 // ddbBucket implements the db interface using dynamodb as the backend
 type ddbBucket struct {
 	ddbBucketStatePrimaryKey
-	// Value is the sum of all increments in the current sliding window for the bucket
-	Value uint `dynamodbav:"value"`
 	// Expiration indicates when the current rate limit expires. We opt not to use DyanamoDB TTLs
 	// because they don't have strong deletion guarantees.
 	// https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/howitworks-ttl.html
 	// "DynamoDB typically deletes expired items within 48 hours of expiration. The exact duration within
 	// which an item truly gets deleted after expiration is specific to the nature of the workload
 	// and the size of the table."
 	Expiration time.Time `dynamodbav:"expiration,unixtime"`
+	// Value is the sum of all increments in the current sliding window for the bucket
+	Value uint `dynamodbav:"value"`
 	// Version is an internal field used to control flushing/draining the Value field concurrently
 	Version uint `dynamodbav:"version"`
+	// TTL is an internal attribute to define how long the item will live in dynamodb prior to being
+	// set for removal. This TTL mechanism is only used for good hygiene to ensure we don't leave
+	// unused buckets in the database forever
+	TTL time.Time `dynamodbav:"_ttl,unixtime"`
 }
 
-func newDDBBucket(name string, expiration time.Time) ddbBucket {
+func newDDBBucket(name string, expiresIn time.Duration, ttl time.Duration) ddbBucket {
 	return ddbBucket{
 		ddbBucketStatePrimaryKey: ddbBucketStatePrimaryKey{
 			Name: name,
 		},
+		Expiration: time.Now().Add(expiresIn),
 		Value:      0,
-		Expiration: expiration,
 		Version:    0,
+		TTL:        time.Now().Add(ttl),
```

Comment:
nit: my OCD gets triggered by these two calls to `time.Now()` and how they'll be slightly different. Can be replaced with one call + storing it in a local variable e.g. `now := time.Now()`